include/sddf/network/queue.h

```diff
@@ -85,7 +85,7 @@ static inline bool net_queue_empty_active(net_queue_handle_t *queue)
  */
 static inline bool net_queue_full_free(net_queue_handle_t *queue)
 {
-    return !((queue->free->tail + 1 - queue->free->head) % queue->size);
+    return !(queue->free->tail + 1 - queue->free->head);
```

Comment:
Both these changes to the full conditions are incorrect. 

Firstly, remember that I haven't yet corrected the issue raised before that if the queue size is `S`, we currently only use `S-1` entries. (This is next on my list of issues to fix.) 

Bearing this in mind, the idea with this function is to catch the case when the `tail` is `S-1` greater than the `head`. E.g.
`tail = head + S - 1`
Re-arranging this we get:
`tail + 1 - head = S`
which satisfies the old condition of:
`!((tail + 1 - head) % S)`

What you have changed it to instead is:
`tail + 1 - head != 0`, or `tail + 1 != head`
which means that a queue is only considered full when the `tail` is `UINT16_MAX - 1` greater than the `head` (much larger than most of our queues :-) ).

I believe the correct condition should be
`tail + 1 - head = size`

Reply:
Yes, my bad. I will fix that.